crates/anvil/tests/it/anvil_api.rs

```diff
@@ -805,6 +805,25 @@ async fn test_reorg() {
     assert!(res.is_err());
 }
 
+#[tokio::test(flavor = "multi_thread")]
+async fn test_rollback() {
+    let (api, handle) = spawn(NodeConfig::test()).await;
+    let provider = handle.http_provider();
+
+    api.mine_one().await; // block height 1
+    api.mine_one().await; // block height 2
+
+    let original_block0_hash =
+        provider.get_block(1.into(), false.into()).await.unwrap().unwrap().header.hash;
+    api.anvil_rollback(RollbackOptions { depth: 1 }).await.unwrap();
+    // assert the chain rolled back to block height 1, and the block hash is kept the same
+    let height = provider.get_block_number().await.unwrap();
+    assert_eq!(height, 1);
+    let new_block1_hash =
+        provider.get_block(1.into(), false.into()).await.unwrap().unwrap().header.hash;
```

Comment:
better to change to `provider.get_block(BlockId::latest(), false.into())` so we know we compare latest block hash. Should we also compare time and other block envs that we update when rollback?

Reply:
good point! 
added this, plus a test case for `None` and better assertions 👌